apps/shared/avifutil.c

```diff
@@ -11,32 +12,35 @@
 #include "avifpng.h"
 #include "y4m.h"
 
-static int32_t calcGCD(int32_t a, int32_t b)
+// |a| and |b| hold int32_t values. The int64_t type is used so that we can negate INT_MIN without
+// overflowing int32_t.
+static int64_t calcGCD(int64_t a, int64_t b)
 {
+    assert(b != 0);
```

Comment:
The actual math here is perfectly reasonable and solid.

About the assert though:

Based on the bug we discussed earlier, we know that the current implementation could allow for one of these values (from user input) to be zero. Instead of asserting here, perhaps just return `0` here and document returning `0` as a failure due to illegal inputs?

We really only call `calcGCD()` in two places in the whole codebase right now. I feel like it'd be better to protect against zeros at those call sites and fail properly instead of just hitting this assert (which would abort/crash).

Reply:
Per our discussion just now, I see that you've already verified that we protect against zeros in the denominators before this function is ever invoked. This change looks good.

Reply:
Done. I checked both files again and figured out a way to remove the assertion.

Now `calcGCD(0, 0)` returns 0, `calcGCD(a, 0)` returns `a`, and `calcGCD(0, b)` returns `b`.

Reply:
> Per our discussion just now, I see that you've already verified that we protect against zeros in the denominators before this function is ever invoked. This change looks good.

Just to clarify: I only verified that we protect against zeros in the denominators before this function is ever invoked in src/avif.c. This is not the case in apps/shared/avifutil.c, so we should not add the assertion in that file. In any case, the second commit in this pull request made this issue moot.